File: 2206. Divide Array Into Equal Pairs/solution.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
 var divideArray = function(nums) {
  let dictionary = {};
  for (let num of nums){
    dictionary[num] = (dictionary[num] || 0) + 1;
  }

  for (let num in dictionary){
    if (dictionary[num] % 2 != 0){
      return false;
    }
  }

  return true;
};

File: 2206. Divide Array Into Equal Pairs/solution.java
class Solution {
  public boolean divideArray(int[] nums) {
    Map<Integer, Integer> dictionary = new HashMap<>();
    for (int index = 0; index < nums.length; index++){
      int num = nums[index];
      if (dictionary.containsKey(num)){
        dictionary.put(num, dictionary.get(num) + 1);
      }
      else{
        dictionary.put(num, 1);
      }
    }

    for (Map.Entry<Integer, Integer> num : dictionary.entrySet()){
      if (num.getValue() % 2 != 0){
        return false;
      }
    }

    return true;
  }
}